inverse_covariance/clean.py

```diff
@@ -0,0 +1,323 @@
+    ----------
+    copy : boolean, optional, default True
+        If False, try to avoid a copy and do inplace scaling instead.
+        This is not guaranteed to always work inplace; e.g. if the data is
+        not a NumPy array or scipy.sparse CSR matrix, a copy may still be
+        returned.
+    with_mean : boolean, True by default
+        If True, center the data before scaling.
+        This does not work (and will raise an exception) when attempted on
+        sparse matrices, because centering them entails building a dense
+        matrix which in common use cases is likely to be too large to fit in
+        memory.
+    with_std : boolean, True by default
+        If True, scale the data to unit variance (or equivalently,
+        unit standard deviation).
+    Attributes
+    ----------
+    scale_ : ndarray, shape (n_features,)
+        Per feature relative scaling of the data.
+        .. versionadded:: 0.17
+           *scale_*
+    mean_ : array of floats with shape [n_features]
+        The mean value for each feature in the training set.
+    var_ : array of floats with shape [n_features]
+        The variance for each feature in the training set. Used to compute
+        `scale_`
+    n_samples_seen_ : int
+        The number of samples processed by the estimator. Will be reset on
+        new calls to fit, but increments across ``partial_fit`` calls.
+    Examples
+    --------
+    >>> from inverse_covariance.clean import TwoWayStandardScaler
+    >>>
+    >>> data = [[1, 0], [1, 0], [2, 1], [2, 1]]
+    >>> scaler = StandardScaler()
+    >>> print(scaler.fit(data))
+    StandardScaler(copy=True, with_mean=True, with_std=True)
+    >>> print(scaler.mean_)
+    [ 3.0  0.5]
+    >>> print(scaler.transform(data))
+    [[-1. -1.]
+     [-1. -1.]
+     [ 1.  1.]
+     [ 1.  1.]]
+    See also
+    --------
+    twoway_standardize: Equivalent function without the estimator API.
+    :class:`sklearn.preprocessing.StandardScaler`
+    :class:`sklearn.decomposition.PCA`
+        Further removes the linear correlation across features with 'whiten=True'.
+    Notes
+    -----
+    See the implications of one-way vs. two-way standardization in here. TBD
+    
+    """  # noqa
+
+    def __init__(self, copy=True, with_mean=True, with_std=True):
+        """Unlike StandardScaler, with_mean is always set to True, to ensure
+        that two-way standardization is always performed with centering. The
+        argument `with_mean` is retained for the sake of sklearn
+        API compatibility.
+        """
+        self.with_mean = True
+        self.with_std = with_std
+        self.copy = copy
+
+    def _reset(self):
+        """Reset internal data-dependent state of the scaler, if necessary.
+        __init__ parameters are not touched.
+        """
+
+        # Checking one attribute is enough, becase they are all set together
+        # in partial_fit
+        if hasattr(self, 'col_scale_'):
+            del self.row_scale_
```

Comment:
Removing `_reset()` as it is an internal function is unused internally and the caller should create a new instance if they do not want to rely on previous state, rather than re-using the transformer.